spec/models/rbac_spec.rb

```diff
@@ -542,6 +542,18 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
           @template2 = FactoryGirl.create(:template_vmware, :name => "Template2", :host => @host_2, :ext_management_system => @ems)
         end
 
+        it "returns all host's VMs and templates when host filter is set up" do
+          @host_1.parent = @hfolder # add host to folder's hierarchy
+          mtc_folder_path_with_host = "#{@mtc_folder_path}/EmsFolder|host/Host|#{@host_1.name}"
+          group.update_attributes(:filters => {"managed" => [], "belongsto" => [mtc_folder_path_with_host]})
+
+          results2 = Rbac.search(:class => "Vm", :user => user, :results_format => :objects).first
+          expect(results2.length).to eq(1)
+
+          results2 = Rbac.search(:class => "VmOrTemplate", :user => user, :results_format => :objects).first
```

Comment:
what about MiqTemplate?  Can it work with provider specific subclasses?  Do we allow them to specify `ManageIQ::Providers::Vmware::InfraManager::Vm` or similar?